# Just some links to software to help analyse pooled sequence data.

## Identifying polymorphisms (that work well for pools.
[Evaluation of variant detection software for pooled next-generation sequence data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4518579/)

[varscan](https://github.com/dkoboldt/varscan). Seems like it has an active community.

[snape](https://www.ncbi.nlm.nih.gov/pubmed/22992255). Not such an active community?


## allele frequencies and basic evolutionary parameters.
[SNPGenie](https://www.ncbi.nlm.nih.gov/pubmed/26227143): estimating evolutionary parameters to detect natural selection using pooled next-generation sequencing data. Link to code on github [here](https://github.com/hugheslab/snpgenie)

[Genotype-Frequency Estimation from High-Throughput Sequencing Data](http://www.genetics.org/content/201/2/473.short). Link to software is [here](https://github.com/Takahiro-Maruki/Package-GFE). 


## Some reasons to be optimistic (or not) about pooled sequencing.
[Validation of SNP allele frequencies determined by pooled next-generation sequencing in natural populations of a non-model plant species.](https://www.ncbi.nlm.nih.gov/pubmed/24244686)

and an or not
[Population-genetic inference from pooled-sequencing data](https://www.ncbi.nlm.nih.gov/pubmed/24787620). The paper "Genotype-Frequency Estimation from High-Throughput Sequencing Data" uses the method discussed here I think.
